File: backend/src/veya/api/schemas/sentiment.py
from pydantic import BaseModel


class AnalyzeSentimentResponse(BaseModel):
    analyzed_comments: int


class SentimentSummaryResponse(BaseModel):
    total: int
    positive: int
    neutral: int
    negative: int
    positive_percentage: float
    neutral_percentage: float
    negative_percentage: float

File: frontend/src/types/sentiment.ts
export type SentimentSummary = {
  total: number;
  positive: number;
  neutral: number;
  negative: number;
  positive_percentage: number;
  neutral_percentage: number;
  negative_percentage: number;
};

export type AnalyzeSentimentResult = {
  analyzed_comments: number;
};
